Top100/top100songs.py

```diff
@@ -0,0 +1,257 @@
+    and return all the lyrics
+
+    :param song_title: str.
+    :param artist: str
+    :return: lyrics: str. None if no URL was found.
+    """
+    url = get_full_url_to_lyrics(song_title, artist)
+    if url is None:
+        return ''
+
+    lyr_page = get_content_from_url(url)
+    lyrics = lyr_page.find('pre', {'id': 'lyric-body-text'})  # , limit=1)
+    return lyrics.text
+
+
+def count_words_in_text(text: str) -> dict:
+    """Counts the number of unique words in a given string.
+    :param text: str
+    :return: word_count: dict
+    """
+    # Check input validation
+    if text is '':
+        return dict()
+    if type(text) is not str:
+        raise ValueError("Input must be string")
+
+    # Clean rows from Text
+    text.replace("\n", ' ')
+
+    # Insert word without punctuations
+    word_count = {}
+    for word in text.split():
+        word = word.strip(string.punctuation).lower()
+        try:
+            word_count[word] = word_count[word] + 1
+        except KeyError:
+            word_count[word] = 1
+    return word_count
+
+
+def get_top_100_songs() -> dict:
+    """This function gets the top 100 songs as published on the
+    Billboard hot-100 page (https://www.billboard.com/charts/hot-100)
+    :return: songs_dict: dict
+    """
+    url = 'https://www.billboard.com/charts/hot-100'
+
+    content = get_content_from_url(url)
+
+    chart_list = content.find_all('ul', class_="o-chart-results-list-row")
+
+    # Find all ranks, and correlating titles and artists
+    songs_dict = {}
+    for element in chart_list:
+        rank = element.find('span', class_='a-font-primary-bold-l').string.strip('\n\t')
+        song_title = element.find('h3', class_='a-no-trucate').string.strip('\n\t')
+        artist = element.find('span', class_='a-no-trucate').string.strip('\n\t')
+
+        lyrics = get_lyrics(song_title, artist)
+        if rank not in songs_dict:
+            songs_dict[rank] = {'Title': song_title,
+                                'Artist': artist.replace(' &', ','),
+                                'Word count': count_words_in_text(lyrics)}
+        else:
+            raise LookupError(f"parsing error. rank {rank} was already accounted for")
+
+    return songs_dict
+
+
+def merge_all_word_counts(song_word_count: dict, tot_word_count: dict) -> dict:
+    """Sums the word counts of all the various song_count dictionaries
+    :param song_word_count: dict
+    :param tot_word_count: dict
+    :return: tot_word_count: dict
+    """
+    for word in song_word_count.keys():
+        try:
+            tot_word_count[word] = tot_word_count[word] + song_word_count[word]
+        except KeyError:
+            tot_word_count[word] = song_word_count[word]
+    return tot_word_count
+
+
+def merge_dictionaries_by_key(songs_dict: dict, key: any) -> dict:
+    """This function runs through a dictionary with values of type dictionary,
+    and merges the values's values by the key given (value[key]).
```

Comment:
values's values?

Reply:
This function was updated completely, as it was missing the actual code to take care different types of values.
Documentation, and code changed.